provider/awssd/aws_sd.go

```diff
@@ -218,24 +265,29 @@ func (p *AWSSDProvider) ApplyChanges(ctx context.Context, changes *plan.Changes)
 	changes.Delete = append(changes.Delete, deletes...)
 	changes.Create = append(changes.Create, creates...)
 
+	// remove redundant targets from Create and Delete
+	dedupCreate, dedupDelete, err := p.DedupDeletesAndCreates(changes.Create, changes.Delete)
+	if err != nil {
+		return err
+	}
```

Comment:
Maybe this part is not needed. Duplicated actions are created just a couple of lines above with `updatesToCreates`. Isn't it better to modify that instead of removing duplicates later?

Reply:
Done. I still keep `DedupDeletesAndCreates` as a separate function from `updatesToCreates` in order to improve readibility. But now the former function is invoked from within the latter, and is a lot simpler.